codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -244,7 +248,7 @@ let checkAndSetSyncedChains = (~nextQueueItemIsKnownNone=false, chainManager: Ch
       cf.timestampCaughtUpToHeadOrEndblock->Option.isSome && accum
     )
 
-  if allChainsSyncedAtHead {
+  if allChainsSyncedAtHead && shouldSetPrometheusSynced {
```

Comment:
This should not be set after pre registration since the hosted service will infer the indexer as synced then.